tests/unit/test_eventstream.py

```diff
@@ -301,8 +297,13 @@ def test_all_positive_cases():
 def test_negative_cases():
     """Test that all negative cases raise the expected exception. """
     for (encoded, exception) in NEGATIVE_CASES:
-        test_function = raises(exception)(check_message_decodes)
-        yield test_function, encoded, None
+        try:
+            check_message_decodes(encoded, None)
```

Comment:
We may want to be using something like:
```python
with pytest.raises(exception):
    check_message_decodes(encoded, None)
```

That will be a bit easier to understand.

Reply:
I didn’t want to add any more direct pytest dependency, than what’s absolutely necessary (I cannot mark tests as slow with unittest only), so that you won’t get to the same situation when pytest dies as nose just did.

Reply:
I think if we're gonna rewrite the test, we should try to write the final version we want. We already have a hard dependency on pytest for the test suite now, so let's use it here if needed.